libs/hscim/src/Web/Scim/Class/Group.hs

```diff
@@ -122,8 +122,8 @@ class (Monad m, GroupTypes tag, AuthDB tag m) => GroupDB tag m where
   getGroups ::
     AuthInfo tag ->
     Maybe Filter ->
-    Maybe Natural ->
-    Maybe Natural ->
+    Maybe Int32 ->
```

Comment:
The `Natural` was there because a valid startIndex and count are positive. Indeed, zero was still wrong. But since the RFC asks to default any invalid values to 1, then it's going to be some non-fitting type for these fields anyway. Since the rest of the hscim library already uses a regular `Int`, I'd just use it here in the API as well for consistency.

Reply:
I may not understand the robustness argument, but what I simply meant that the API at hand here is within the hscim library, and within this library there is also the data type `ListResponse` which has fields for these same values that the current API definition captures, and those fields already have the `Int` type. Hence I'd also use regular `Int`s in the API as well because they represent the same thing.\

(It could also be that you already knew all that and I'm just missing the point :p)